iroh-net/src/relay/server/actor.rs

```diff
@@ -73,37 +69,30 @@ pub struct ServerActorTask {
     // TODO: stats collection
 }
 
-impl ServerActorTask {
-    /// TODO: replace with builder
-    pub fn new(key: SecretKey) -> Self {
+impl Default for ServerActorTask {
+    fn default() -> Self {
         let (server_channel_s, server_channel_r) = mpsc::channel(SERVER_CHANNEL_SIZE);
-        let server_actor = ServerActor::new(key.public(), server_channel_r);
+        let server_actor = ServerActor::new(server_channel_r);
         let cancel_token = CancellationToken::new();
         let done = cancel_token.clone();
         let server_task = AbortOnDropHandle::new(tokio::spawn(
-            async move { server_actor.run(done).await }
-                .instrument(info_span!("relay.server", me = %key.public().fmt_short())),
+            async move { server_actor.run(done).await }.instrument(info_span!("relay.server")),
         ));
-        let meta_cert = init_meta_cert(&key.public());
+
         Self {
             write_timeout: Some(WRITE_TIMEOUT),
-            secret_key: key,
-            meta_cert,
             server_channel: server_channel_s,
             closed: false,
             loop_handler: server_task,
             cancel: cancel_token,
         }
     }
+}
 
-    /// Returns the server's secret key.
-    pub fn secret_key(&self) -> &SecretKey {
-        &self.secret_key
-    }
-
-    /// Returns the server's public key.
-    pub fn public_key(&self) -> PublicKey {
-        self.secret_key.public()
+impl ServerActorTask {
+    /// Creates a new default `ServerActorTask`.
+    pub fn new() -> Self {
+        Self::default()
```

Comment:
one tiny bit of nitpicking/bikeshedding: I usually prefer to have the code in `new` than `default` as that's more obvious when viewing source on docs.rs.  but either way works.